themes/default/content/blog/benchmarking-python-performance/index.md

```diff
@@ -0,0 +1,216 @@
+we do record the status code of each command so can show they executed successfully.
+
+## Python: Pre- and Post-patch
+
+This experiment compares the performance of Pulumi Python before and after
+the patch was applied. The control group used Pulumi v3.43.1, while the
+experimental group used Pulumi v3.44.3. The primary difference between these
+two groups is that a fix was introduced for a Python runtime concurrency bug
+as part of v3.44.0. Both groups use the same benchmark program, which created
+100 AWS SNS and 100 AWS SQS resources in parallel, as described earlier. Only
+the version of the Pulumi CLI is different between groups.
+
+### [Control vs. Fix](https://app.warp.dev/block/rk7fFf2jn2iKXYcIXwhZ8F)
+
+| **Group**        | **Mean**  | **Standard Deviation** |
+| ---------------- | --------- | ---------------------- |
+| **Control**      | 222.232 s | 0.908 s                |
+| **Experimental** | 70.189 s  | 1.497 s                |
+
+**Summary:** The  **Experimental Group** ran 3.17 ± 0.07 times faster than the **Control Group**, accounting for a +300% speedup in performance. Running Welch T-Test indicated statical significance (p = 2.93e-59, α=0.05).
+
+## Python vs. TypeScript
+
+After seeing very promising results from the first experiment, we wanted to
+determine just how promising these results were. We decided to compare Pulumi
+Python to Pulumi TypeScript to see if this fix had narrowed the gap in
+performance between the two runtimes. We ported the Python program to TypeScript:
+
+```typescript
+import * as pulumi from "@pulumi/pulumi";
+import * as aws from "@pulumi/aws";
+
+// SQS
+[...Array(100).map((_, i) => {
+  const name = `pulumi-${i}`;
+  new aws.sqs.Queue(name);
+})];
+
+// SQS
+[...Array(100).map((_, i) => {
+  const name = `pulumi-${i}`;
+  new aws.sqs.Queue(name);
+})];
+```
+
+For this experiment, we fixed the version of the CLI to v3.44.3, which included
+the patch to the Python runtime. Here are the result.
+
+### [TypeScript vs. Python](https://app.warp.dev/block/rk7fFf2jn2iKXYcIXwhZ8F)
+
+| **Group**      | **Mean** | **Standard Deviation** |
+| -------------- | -------- | ---------------------- |
+| **Python**     | 70.975 s | 0.909 s                |
+| **TypeScript** | 73.741 s | 1.574 s                |
+
+**Summary:** The **Python Group** performed the best and ran 1.04 ± 0.03 times
+faster than the **TypeScript Group**. This accounts for a 4% difference in
+performance. A second T-Test indicated statical significance
+(p = 1.4e-07, α=0.05). Not only did Python close the gap with TypeScript,
+but it is also now marginally faster than its Node.js competitor.
+
+## Conclusion
+
+It's rare to have a small PR result in such a massive performance increase,
+but when it happens, we want to shout it from the rooftops. This change, which
+shipped last year in v3.44.3, does not require Python users to opt-in; their programs
+should just be faster. This patch has closed the gap with the Node.js runtime.
```

Comment:
Small suggestion, non blocking; since the change shipped (and we're confident about the performance) last year; should "should just be faster" be changed to "are now faster"

Reply:
Fixed. Thanks!